### PR TITLE
remove Makefile on dev-asm

### PR DESCRIPTION
**Pull request by gogo2464**
Fixes https://github.com/rizinorg/book/issues/97

Focuses on asm plugin only. Does not document meson on other plugins.

### CODE REVIEW

File: src/plugins/dev-asm.md [hunk 1]
```diff
@@ -24,32 +24,6 @@ and length:
...
 
-**Makefile**
-
-```makefile
-NAME=asm_mycpu
-RZ_PLUGIN_PATH=$(shell rizin -H RZ_USER_PLUGINS)
-LIBEXT=$(shell rizin -H LIBEXT)
-CFLAGS=-g -fPIC $(shell pkg-config --cflags rz_analyss)
-LDFLAGS=-shared $(shell pkg-config --libs rz_analysis)
-OBJS=$(NAME).o
```
[the remark this answers is not in the document]
Comment (gogo2464) [reply src/plugins/dev-asm.md:35]: I belived it has been removed in favor of meson.

[the remark this answers is not in the document]
Comment (XVilka) [reply src/plugins/dev-asm.md:35]: Instead of just removing it's better to provide the example of the Meson file

**Review by XVilka — CHANGES_REQUESTED** [review]

File: src/plugins/dev-asm.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (wargio) [reply src/plugins/dev-asm.md:35]: > the makefile has been removed

This is a misconception! the book entry is about making an asm plugin for rizin, and to build the plugin you can use a Makefile. I do agree we have removed them from rizin itself in favor of meson, but is not strictly required to use meson also for the external plugins.

That said, i strongly agree with xvilka to have this replaced with a `meson.build` example is more beneficial.

[the remark this answers is not in the document]
Comment (gogo2464) [reply src/plugins/dev-asm.md:35]: I have already a branch of rizin with asm plugin with meson. I can update the documentation for meson.

Should I remove Makefile doc and replace it with meson or just add meson? Two way might be confusing.

[the remark this answers is not in the document]
Comment (wargio) [reply src/plugins/dev-asm.md:35]: remove the makefile and add the meson example.

[the remark this answers is not in the document]
Comment (gogo2464) [reply src/plugins/dev-asm.md:35]: Perfect! I know what I should do. I should be able to fix it this week.

Sorry for the time I made you waste in this misundertsanding.

[the remark this answers is not in the document]
Comment (wargio) [reply src/plugins/dev-asm.md:35]: doesn't matter. better wasting some time and have a good PR.

File: src/plugins/dev-asm.md [hunk 2]
```diff
@@ -127,6 +101,7 @@ The build is handled by meson, we have to add our plugin to the list of things t
    #'arch/6502/6502dis.c',
    'arch/amd29k/amd29k.c',
    #'arch/8051/8051_disas.c',
++  ''arch/mycpu/mycpu_disas.c,
```
Comment (wargio) [inline src/plugins/dev-asm.md:104]: this is wrong.
```suggestion
+  'arch/mycpu/mycpu_disas.c',
```

**Comment by gogo2464** [comment]
Seems good for me.

Sadly I have not really compiled the mycpu plugin really.
Hopefully I have created my own plugin to handle evm: https://github.com/rizinorg/rizin/pull/3595 so I can see.

**Comment by wargio** [comment]
can you add to the example the meson template?

**Comment by gogo2464** [comment]
I believed I have already done https://github.com/rizinorg/book/pull/98/files#diff-cc0314ce2a2ce993dd86f5b416f5ece4ad633a55911c53f41c1b23a6ae688b16R89

**Comment by gogo2464** [comment]
@wargio Did I forgot something?

**Review by wargio — APPROVED** [review]
Sorry, i misread it yesterday.

**Comment by gogo2464** [comment]
Seems ready to merge @XVilka  ?

**Review by XVilka — CHANGES_REQUESTED** [review]
You added how to add it to a Rizin tree, but please also add a standalone Meson file, like it was done with Makefile before. You can see examples from rizin-extras how it can be done.

**Comment by gogo2464** [comment]
@XVilka is it better now?

**Review by XVilka — CHANGES_REQUESTED** [review]
No, what I meant, is you just need a meson file for just one particular plugin, just like this: https://github.com/rizinorg/rizin-extras/blob/master/jaguar-gpu/meson.build

**Comment by gogo2464** [comment]
I am confused... like this?

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by gogo2464** [comment]
@wargio @XVilka Does it pass GH-PAGES? I got a warning by email. :-/

**Pull request merged**